src/Sarif.Driver/Sdk/SkimmerBase.cs

```diff
@@ -23,7 +23,9 @@ public SkimmerBase()
 
         abstract protected ResourceManager ResourceManager { get; }
 
-        abstract protected IEnumerable<string> TemplateIds { get; }
+        abstract protected IEnumerable<string> TemplateResourceIds { get; }
```

Comment:
IMO this is a clearer name. These are not in fact the SARIF template ids; they are resource ids of the form `<ruleId>_<templateId>`, from which the template ids are extracted.